Add a read-only checkbox

matthiasclasen opened the pull request.

matthiasclasen committed:
Add files to the document portal

Call out to the document portal and return uris pointing
to the fuse location. For now, we always request read, write
and grant privileges.

matthiasclasen committed:
Build a proxy for the documents portal

This will be used in the next commit.

matthiasclasen committed:
Set up filter filters when they are sent

GTK+ will send filters as (sa(us)) where the first string is
the name of the filter, and the array contains the filter rules
with type 0 for patterns and type 1 for mime types.

matthiasclasen committed:
Use some more fields that get sent by GTK+

Use the current_name, current_folder and current_file fields
if they are present.

matthiasclasen committed:
Add a read-only toggle

This is something where we can take advantage of the sandboxing
and offer users a new feature: We can give applications read-only
access to files.

alexlarsson commented:
Also, we should have a flag in the request to say that we only need a readonly version, because in that case we can simplify the ui by not having this checkbox.

matthiasclasen commented:
good idea, I'll add that

matthiasclasen committed:
Add a way to request readonly or writable file access

If applications don't need write access, we don't have to show
the readonly checkbox and can just export the file without
write access.
